src/compiler/scala/tools/nsc/ast/parser/Parsers.scala

```diff
@@ -872,6 +872,10 @@ self =>
         case Parens(args) => mkNamed(args)
         case _            => right :: Nil
       }
+      def msg(what: String) = s"""multi-parameter infix notation is $what; use dot-and-parenthesis instead"""
```

Comment:
I see you have auto-triple-quote enabled in your IDE. I wasn't sure about "d-and-p", but any other phrase such as "ordinary function invocation syntax" is not much help. Since you already use curly brace, I'd have put `msg` in that block, but of course NBD.

Quick note on "multi-parameter infix": Isn't a binary op always multi-param?, I can hear myself saying. That is, it's multi-operand; multi-param refers to the desugaring. It could say it the other way around: using infix with anything except two operands is deprecated; use ordinary function application syntax instead, such as `x.f(y, z)`.

Reply:
> I see you have auto-triple-quote enabled in your IDE.

I don't use IDE, but maybe I should starting using one.

> Since you already use curly brace, I'd have put `msg` in that block, but of course NBD.

I feel like I've heard somewhere that some in Scala team doesn't like nested functions, but I'd be happy to move `msg` into `if` block.

> using infix with anything except two operands is deprecated

You can write `1 + 2 * 3`, which involves three operands but valid infix notation. How about this?

"using a multi-argument parameter list in an infix notation is deprecated; use ordinary function application syntax instead, such as `x.f(y, z)`"

Reply:
I wish I'd had a good comeback for `1 + 2 * 3` as binary operands twice as opposed to three. (It clearly isn't a ternary anything.) For some reason, Uncle Remus just popped into my head, Zippity doo-dah, etc, Mr Bluebird's on my shoulder. Because of zippity and zip operator?